src/integrations/front-end/feed-improvements.php

```diff
@@ -81,19 +82,48 @@ public function filter_bloginfo_rss( $show, $what ) {
 	}
 
 	/**
-	 * Adds a canonical link header to the main canonical URL for the requested feed object.
+	 * Makes sure send canonical header always runs, because this RRS hook does not support the for_comments parameter
+	 *
+	 * @return void
+	 */
+	public function handle_rss_feed() {
+		$this->send_canonical_header( false );
+	}
+
+	/**
+	 * Adds a canonical link header to the main canonical URL for the requested feed object. If it is not a comment
+	 * feed.
+	 *
+	 * @param bool $for_comments If the RRS feed is meant for a comment feed.
+	 *
+	 * @return void
 	 */
-	public function send_canonical_header() {
-		if ( \headers_sent() ) {
+	public function send_canonical_header( $for_comments ) {
+
+		if ( $for_comments || \headers_sent() ) {
 			return;
 		}
 
 		$url = $this->get_url_for_queried_object( $this->meta->for_home_page()->canonical );
 		if ( ! empty( $url ) ) {
+
 			\header( \sprintf( 'Link: <%s>; rel="canonical"', $url ), false );
 		}
 	}
 
+	/**
+	 * Adds noindex, follow tag for comment feeds.
+	 *
+	 * @param bool $for_comments If the RRS feed is meant for a comment feed.
+	 *
+	 * @return void
+	 */
+	public function add_robots_headers( $for_comments ) {
+		if ( $for_comments && ! \headers_sent() ) {
+			\header( 'x-robots-tag: noindex, follow', true );
```

Comment:
It's probably minor, but docs and our codebase have this tag as capitalized, I'm fixing it also
```suggestion
			\header( 'X-Robots-Tag: noindex, follow', true );
```